src/core/database/repositories/viagemRepository.ts

```diff
@@ -31,10 +31,27 @@ export class ViagemRepository implements IViagemRepository {
     return await this.repository.save(viagemOriginal);
   }
 
-  async excluir(id: number): Promise<boolean> {
-    const viagemExistente = await this.repository.findOneOrFail({ where: { id } });
+  async obterTodasComEntradasEImagens(): Promise<ViagemModel[]> {
+    const result = await this.repository
+      .createQueryBuilder('viagem')
+      .addSelect('CAST(viagem.data AS DATE) as dataFormatada')
+      .leftJoinAndSelect('viagem.entradas', 'entrada')
+      .leftJoinAndSelect('entrada.imagens', 'imagem')
+      .groupBy('viagem.id, entrada.id') // Agrupa por IDs para evitar duplicatas
+      .addSelect(['imagem.id', 'imagem.caminho']) // Seleciona os campos desejados da imagem
+      .addSelect('COUNT(imagem.id) as totalImagens') // Conta o total de imagens por entrada
+      .orderBy('entrada.id', 'ASC') // Ordena por ID de entrada para pegar apenas a primeira
```

Comment:
Aqui deveria ordenar por data em ordem decrescente

```suggestion
      .orderBy('entrada.data', 'DESC') 
```